pkg/util/mount/mount_linux.go

```diff
@@ -31,28 +32,97 @@ import (
+	expectedNumFieldsPerLine = 6
+)
 
 // Mounter implements mount.Interface for linux platform.
 type Mounter struct{}
 
-// Mount wraps syscall.Mount()
-func (mounter *Mounter) Mount(source string, target string, fstype string, flags uintptr, data string) error {
-	glog.V(5).Infof("Mounting %s %s %s %d %s", source, target, fstype, flags, data)
-	return syscall.Mount(source, target, fstype, flags, data)
+// Mount mounts source to target as fstype with given options. 'source' and 'fstype' must
+// be an emtpy string in case it's not required, e.g. for remount, or for auto filesystem
+// type, where kernel handles fs type for you. The mount 'options' is a list of options,
+// currently come from mount(8), if no more option is required, call Mount with an empty
+// string list or nil.
+func (mounter *Mounter) Mount(source string, target string, fstype string, options []string) error {
+	// The remount options to use in case of bind mount. It equals:
+	//   options - 'bind' + 'remount' (no duplicate)
+	bindRemountOpts := []string{"remount"}
+	bind := false
+
+	if options != nil {
+		for _, option := range options {
+			switch option {
+			case "bind":
+				bind = true
+				break
+			case "remount":
+				break
+			default:
+				bindRemountOpts = append(bindRemountOpts, option)
+			}
+		}
+	}
+
+	if bind {
+		err := doMount(source, target, fstype, []string{"bind"})
+		if err != nil {
+			return err
+		}
+		return doMount(source, target, fstype, bindRemountOpts)
+	} else {
+		return doMount(source, target, fstype, options)
+	}
 }
 
-// Unmount wraps syscall.Unmount()
-func (mounter *Mounter) Unmount(target string, flags int) error {
-	return syscall.Unmount(target, flags)
+func doMount(source string, target string, fstype string, options []string) error {
+	glog.V(5).Infof("Mounting %s %s %s %v", source, target, fstype, options)
+	// Build mount command as follows:
+	//   mount [-t $fstype] [-o $options] [$source] $target
+	mountArgs := []string{}
+	if len(fstype) > 0 {
+		mountArgs = append(mountArgs, "-t", fstype)
+	}
+	if options != nil && len(options) > 0 {
+		mountArgs = append(mountArgs, "-o", strings.Join(options, ","))
+	}
+	if len(source) > 0 {
+		mountArgs = append(mountArgs, source)
+	}
+	mountArgs = append(mountArgs, target)
+	command := exec.Command("mount", mountArgs...)
+	output, err := command.CombinedOutput()
+	if err != nil {
+		glog.Errorf("Mount failed: %v\nMounting arguments: %s %s %s %v\nOutput: %s\n",
+			err, source, target, fstype, options, string(output))
+	}
+	return err
 }
 
-// How many times to retry for a consistent read of /proc/mounts.
-const maxListTries = 3
+// Unmount unmounts target with given options.
+func (mounter *Mounter) Unmount(target string, options []string) error {
+	glog.V(5).Infof("Unmounting %s %v", target, options)
+	unmountArgs := []string{}
+	if options != nil {
+		// If options is an empty list, this is a no-op.
+		unmountArgs = append(unmountArgs, options...)
```

Comment:
in Mount() you treat options as options, and here you treat them as flags.  I think you should still treat them as options (e.g. "lazy" rather than "--lazy") and transform them into commandlines in here.